### PR TITLE
[CI] move distributed test into its own CI job

### PR DESCRIPTION
**Pull request by walterddr**
Moving distributed to its own job. 

- [x] ensure there should be a distributed test job for every default test job matrix (on GHA) 
- [x] ensure that circleci jobs works for distributed as well
- [x] waiting for test distributed to have its own run_test.py launch options, see #63147

### CODE REVIEW

**Review by seemethere — APPROVED** [review]

File: .circleci/cimodel/data/pytorch_build_definitions.py [hunk 1]
```diff
@@ -415,6 +415,26 @@ def instantiate_configs(only_slow_gradcheck):
...
 
+        if (
+            compiler_name != "clang"
+            and not is_libtorch
+            and not is_vulkan
+            and not is_pure_torch
+            and not is_noarch
+            and not is_slow_gradcheck
+            and not only_slow_gradcheck
+        ):
```
Comment (seemethere) [inline .circleci/cimodel/data/pytorch_build_definitions.py:426]:
```suggestion
        if compiler_name != "clang"
           and not any(
               is_libtorch,
               is_vulkan
               is_pure_torch
               is_noarch
               is_slow_gradcheck
               only_slow_gradcheck
        ):
```

nit to make it a bit more readable (although this isn't really that much more readable admittedly)

**Review by seemethere — APPROVED** [review]

File: .circleci/cimodel/data/pytorch_build_definitions.py (hunk 1 above)
Comment (walterddr) [reply .circleci/cimodel/data/pytorch_build_definitions.py:426]: yeah it isn't that much more readable, how about let's just follow all other if statement format for now

File: .github/scripts/generate_ci_workflows.py [hunk 2]
```diff
@@ -139,6 +141,9 @@ def __post_init__(self) -> None:
         if not self.on_pull_request:
             self.only_build_on_pull_request = False
 
+        if self.distributed_test:
+            self.enable_distributed_test = True
```
Comment (walterddr) [inline .github/scripts/generate_ci_workflows.py:145]:
```suggestion
            self.enable_distributed_test = 1
```

File: .github/scripts/generate_ci_workflows.py [hunk 3]
```diff
@@ -138,10 +138,12 @@ class CIWorkflow:
     only_build_on_pull_request: bool = False
     only_run_smoke_tests_on_pull_request: bool = False
     num_test_shards_on_pull_request: int = -1
+    distributed_test: bool = True
```
Comment (walterddr) [inline .github/scripts/generate_ci_workflows.py:141]: currently set as default true just to make sure we don't exclude any existing tests we are currently doing

**Pull request closed without merging**